python/lsst/pipe/base/executionButlerBuilder.py

```diff
@@ -265,7 +276,11 @@ def buildExecutionButler(butler: Butler,
     if not outputLocation.isdir():
         raise NotADirectoryError("The specified output URI does not appear to correspond to a directory")
 
-    exports, inserts = _accumulate(graph)
+    # Gather all DatasetTypes from the Python and check any that already exist
+    # in the registry for consistency.
```

Comment:
Are you relying on the registry get datasettype method to raise if it is not found in the registry for this comments behavior?

Reply:
Well, it does call `Registry.getDatasetType`, but it actually catches that exception and proceeds if the missing dataset type is an output, because we want to allow the user to only register the dataset types in the main repo later.

Reply:
This seems inconsistent with the comment that is here. I saw the exception catching, and was thinking there must be something else going on based on this comment. This also leads to the question, how are we propagating the info if --register-dataset-types is passed to know if they should be inserted at the end.

I personally think we should check and throw if they are not in the registry and --register-dataset-types is not in args, but not actually do the registration until the end. This will prevent the case where someone mistypes a name, and they find out at the beginning instead of the end when they have a weird type.

Reply:
Adjusted the comment to clarify and created https://jira.lsstcorp.org/browse/DM-30845.

Reply:
I'm pretty sure that if someone is creating an execution butler they want the "register-dataset-types" to apply to the execution butler and not the upstream butler -- the question is whether having both options is fatal, or does the right thing and ignores main registry, or is really taken to create in main registry. Are we explicitly saying that if you have the execution butler option set that we are guaranteeing that main registry will never be touched (so writeable=False for that butler)

Reply:
The user cannot have a choice about whether to register dataset types in the execution butler - if they don't, it will fail.  So there's no option for that at present, and that's reasonable.

And yes, we do guarantee that if you're *just* making an execution butler, the main registry will never be touched (via `writeable=False`).  That's true more generally of `pipetask qgraph`, which is the subcommand that BPS should be using to do that.  I think it's possible to ask `pipetask run` to make an execution butler but that feels like a mistake in the click stuff, even if it even works.